sata.go

```diff
@@ -745,3 +761,29 @@ func computeAttributeRawValue(mapping ataDeviceAttr, vendorBytes [6]byte, reserv
 	}
 	return rawValue
 }
+
+type AtaSmartThresholdsPage struct {
+	Revnumber  uint16
+	Thresholds map[uint8]uint8
+}
+
+func (d *SataDevice) ReadSMARTThresholds() (*AtaSmartThresholdsPage, error) {
+	pageRaw, err := d.readSMARTThresholds()
+	if err != nil {
+		return nil, err
+	}
+
```

Comment:
we need to check `pageRaw.Checksum` here to ensure the incoming data is not corrupted.

Reply:
I'm trying to find the proper algorithm to calculate the `Checksum` but... I'm not sure.
We need a `uint8`(`byte`) checksum.
What I found:
* github.com/go-daq/crc8 - doesn't allow getting correct checksum with any Table (tried all possible polynomial 0..255)
* github.com/howeyc/crc16 - allows getting correct checksum (in the data I have it should be `233` with the polynomial equal to `177`) after mod `0x0100`:
```
	table := crc16.MakeTable(uint16(177))
	checksum := crc16.Checksum(thresholdsByte[:511], table) % 0x0100
```
* manual calculation with the following algorithm:
```
	var ch uint16
	for i := 0; i < 511; i++ {
		ch = ch + uint16(respBuf[i])
	}
	checksum := 0x0100 - ch%0x0100
```
However, the second option seems just a heuristic, I have no arguments for the `177` polynomial, the third option was found in Internet, but again I have no arguments to use exactly it.

Reply:
Could you check this approach:
```
	var ch uint16
	for i := 0; i < 511; i++ {
		ch = ch + uint16(respBuf[i])
	}
	checksum := 0x0100 - ch%0x0100
```
On your host, since unfortunately I don't have another device with SATA SSD...